pkg/client/client.go

```diff
@@ -27,23 +37,20 @@ type Client interface {
 	// * func()
 	// * func() error
 	// * func(context.Context)
-	// * func(context.Context) error
+	// * func(context.Context) event.Result
 	// * func(event.Event)
-	// * func(event.Event) error
+	// * func(event.Event) event.Result
 	// * func(context.Context, event.Event)
-	// * func(context.Context, event.Event) error
-	// * func(event.Event, *event.EventResponse)
-	// * func(event.Event, *event.EventResponse) error
-	// * func(context.Context, event.Event, *event.EventResponse)
-	// * func(context.Context, event.Event, *event.EventResponse) error
+	// * func(context.Context, event.Event) event.Result
```

Comment:
I think this interface is good also to check if a `transport` implementation effectively supports the request/response model. When you set a function which can return a result, BUT the transport doesn't support writing "responses", this function fails